### PR TITLE
Create a single SHA for patch releases

### LINKED ISSUE #28535: Prepare a single branch with all the patch sources correctly updated.

### PR DESCRIPTION
**Pull request by pallavit**
1. This script generates a single SHA for patch releases - with the changelog, readme and other sources in the releasable state
2. Creates a separate branch for the BOM releases - with the snapshot of the version_client.txt file.
3. Generates a report containing the name of the branches (one for BOM and other for artifacts) and the name of each artifact that needs to be patched.

Fixes #28535

### CODE REVIEW

**Review by JimSuplizio — APPROVED** [review]
Thanks for making these updates @pallavit.

File: eng/pipelines/bompreparation.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+trigger: none
+
+pr:
+  branches:
+    include:
+      - main
+  paths:
+    include:
+      - eng/pipelines/bompreparation.yml
```
Comment (JimSuplizio) [inline eng/pipelines/bompreparation.yml:9]:
```suggestion
pr: none
```
We don't want this ever being triggered other than manually.

**Review by JimSuplizio — CHANGES_REQUESTED** [review]

**Comment by pallavit** [comment]
@JimSuplizio I have updated the pipeline trigger. Let me know if you have any other concerns. I have also set the PR to auto-merge so unless you have any other comments, your approval will merge this PR.
/cc: @vcolin7, @srnagar

**Review by JimSuplizio — APPROVED** [review]
Thank you @pallavit for updating the triggers. This isn't a pipeline we want running unless we explicitly run it.

**Pull request merged**